youtube_dl/extractor/giga.py

```diff
@@ -7,13 +7,47 @@
 from ..utils import (
     qualities,
     compat_str,
-    parse_duration,
     parse_iso8601,
-    str_to_int,
+    int_or_none,
+    float_or_none,
 )
 
 
 class GigaIE(InfoExtractor):
+    _VALID_URL = r'https?://videos\.giga\.de/embed/(?P<id>\d+)'
+
+    def _call_api(self, path, video_id):
+        return self._download_json('http://www.giga.de/api/syndication/video/video_id/%s/%s.json' % (video_id, path), video_id)
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        default = self._call_api('default', video_id)
+        playlist = self._call_api('playlist', video_id)[0]
+
+        quality = qualities(['normal', 'hd720'])
+
+        formats = []
+        for format_id in itertools.count(0):
```

Comment:
> Isn't 0 already the default?

i didn't change the code in this part.
it's already in the master branch https://github.com/rg3/youtube-dl/blob/master/youtube_dl/extractor/giga.py#L61